content/client-lib-development-guide/features.textile

```diff
@@ -378,14 +383,19 @@ h3(#realtime-channel). Channel
 * @(RTL11)@ If a channel enters the @DETACHED@ or @FAILED@ state, then all messages that are still queued for send on that channel should be deleted from the queue triggering a failure for the publish or presence methods invoked for those messages
 * @(RTL4)@ @Channel#attach@ function:
 ** @(RTL4a)@ If already @ATTACHED@ or @ATTACHING@ nothing is done
+** @(RTL4e)@ If the channel is in the @DETACHING@ state, it will result in an error, as the channel must complete the detach request before reattaching
```

Comment:
I am not sure this is a good idea as it makes it a bit tricky for a developer as they need to be sure the detach is complete. Why not simply: `If the channel is in the @DETACHING@ state, the attach operation is executed once the channel enters the @DETACHED@ state, or results in an error if it moves to the @FAILED@ state`